### DISCUSSION TITLE
v0.17.0

### DISCUSSION THREAD

**Discussion by nielsvanvelzen** [jellyfin/jellyfin-androidtv#3822] [General]
The latest Android TV app release features enhanced stability, improved navigation with a new home button, and an updated screensaver with age rating filters. Read the [blog post](https://jellyfin.org/posts/androidtv-v0.17.0) to learn about the changes in this release, or read the changelog below.

## 🌟 Highlights

- Add home button when away from home view #3262, by [@VizuaaLOG](https://github.com/VizuaaLOG)
- Add preferences for screensaver age rating #3513, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Implement normalization gain for music tracks #3660, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add media logo to screensaver #3680, by [@haydenzeller](https://github.com/haydenzeller)

## 🏗️ Enhancements

- Watched indicator settings string clarity improvements #2997, by [@DoggoOfSpeed](https://github.com/DoggoOfSpeed)
- Add ability to sort by runtime #3084, by [@bathompson](https://github.com/bathompson)
- Add more item types to search #3266, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Support setting audio and subtitle streams via remote control #3288, by [@3l0w](https://github.com/3l0w)
- Show specials for seasons #3292, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Show specials for series #3301, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove "auto" max bit rate and add additional values #3372, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite BaseItemInfoRow with Compose #3379, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove "automatically choose" option from video player preference #3407, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add "clear image cache" button to developer preferences #3409, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add fallthrough to media3 MimeTypes in codec\-\>mime type mapping #3410, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Enable Vorbis audio in video direct play #3417, by [@MichaelRUSF](https://github.com/MichaelRUSF)
- Remove non\-profile usages of DeviceUtils #3425, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Improve info row colors #3448, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Update to Kotlin SDK 1\.5\.0\-beta\.1 \(Jellyfin 10\.9\) #3472, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Update SDK to 1\.4\.7 \(10\.8 with enum backports\) #3478, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add button to pick a random item to the smart screen #3487, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Split playback preferences into normal and advanced screens #3497, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove leakcanary #3523, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Refactor QueueEntry design #3554, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Cache computed PlayableMediaStream in QueueEntry element #3555, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Initial video support in playback rewrite #3557, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Initial subtitle support in playback rewrite #3564, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add initial video codec testing #3565, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add media source selection #3566, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove unused "Offline" item location #3603, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate to androidx\.tv\.material #3611, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Reordering the trailer button #3648, by [@efreet7](https://github.com/efreet7)
- Change 'Queue all subsequent episodes' setting to queue all episodes in a series #3658, by [@ConnorS1110](https://github.com/ConnorS1110)
- Simplify video player preference #3663, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add a blur to backdrop images #3674, by [@haydenzeller](https://github.com/haydenzeller)
- Use ViewModel in DreamService #3692, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove hasPassword support #3710, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Shrink BaseItemInfoRowRuntime size #3724, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite details row with constraint layout #3729, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add direct play support for WAV audio #3737, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Improve BlurHash decoding performance #3753, by [@cbeyls](https://github.com/cbeyls)
- Increase ExoPlayer HTTP timeouts #3759, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Update editor config #3760, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Update debug media details #3783, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove unused dtsEnabled preference #3785, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add PCM\_S16LE\, PCM\_S20LE and PCM\_S24LE to supported audio codecs #3788, by [@Bond\-009](https://github.com/Bond\-009)
- Update targetSdk to 34 #3807, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 💥 Crash fixes

- Fix ModelCompat crash for media type #3607, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix app crash in EditServerScreen #3615, by [@pascalwei](https://github.com/pascalwei)
- Fix JSON decode crash in PictureViewerFragment #3627, by [@pascalwei](https://github.com/pascalwei)
- Fix crash when re\-opening app during music playback #3713, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Catch errors in ExoPlayerAudioPipeline #3716, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix crash when refreshing current item #3723, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix crash when item has no media sources #3731, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Refactor Live TV popups #3733, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix various Live TV issues #3743, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix nullability in copyWith\[Series\]TimerId functions #3803, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 🔧 Bugfixes

- android tv hls playback direct play fix #3348, by [@mnsrulz](https://github.com/mnsrulz)
- Specify BUILD\_TOOLS\_VERSION for r0adkll/sign\-android\-release step in publish workflow #3369, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix screensaver and media manager displaying video entries #3558, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Refactor BaseRowItem to use inheritance #3598, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix resumable episodes showing up twice in "Play next" integration #3625, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix unchecked warning in FullDetailsFragment #3629, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix login with pruned sessions #3632, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Start series from first unwatched in "play next up" #3633, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix now playing fragment not using string resource for current queue title #3638, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix start letter & favorite playlist implementations for item row #3653, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix QuickConnect sticky login state #3656, by [@pascalwei](https://github.com/pascalwei)
- Fix play after pause in MediaManager implementation #3661, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Various small bug fixes #3666, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix version selection #3677, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Improve "continue listening" row #3684, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix Live TV not playing video streams #3687, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use Enum\.name instead of Enum\.toString in preference stores #3707, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix skip to previous/next playback actions #3708, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix header position in BrowseFolderFragment #3709, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix PlaySessionService no longer reporting to server #3712, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix refreshing current playing item hiding play icon #3715, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix isPlayingAudio behavior #3720, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix authorization for transcoding #3735, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix "Next episode\(s\)" row not showing #3746, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix race condition in player backdrop #3751, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix AudioNowPlayingFragment showing "From null" when there is no album name #3752, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix timezone issues in Live TV guide #3761, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix FullDetailsFragment duplicating buttons on resume #3774, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Disable tunneling #3786, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix NPE when opening Live TV guide for the first time #3790, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix "other" items in ByLetterFragment #3802, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix record buttons in LiveProgramDetailPopup swapped #3808, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix Live TV program popup not populating when opening early #3809, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 🔃 Refactoring

- Remove unnecessary OptIn annotations #3235, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove legacy media manager implementation #3321, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Simplify MediaManager interface #3322, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove baseline workaround #3333, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use data object when applicable #3350, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Replace deprecated rememberKoinInject with koinInject #3408, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove non existent "SeriesTimers" collection type #3493, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use UUID for "itemId" parameters #3494, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove PlayAccess checks #3507, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use androidx fragment\-compose #3549, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate from java\.util\.Date to java\.time #3550, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Replace PlaybackForwardingActivity with PlaybackRewriteFragment #3559, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use dependency injection for ItemLauncher\, KeyProcessor and ReportingHelper #3572, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fully migrate ItemListFragment to use SDK #3574, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove FakeBaseItem hack #3581, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Refactor PlaybackHelper to use dependency injection #3584, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite PlaybackHelper to use the SDK #3586, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove ImageUtils #3587, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove unused code branch in BrowseViewFragment #3589, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Refactor FullDetailsFragment to use SDK #3591, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate "next up" row to SDK #3624, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove some unused code related to ItemRowAdapter #3637, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove LibVLC #3639, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate "Recently added" to SDK #3640, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate various item rows to SDK #3643, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate Live TV rows to SDK #3644, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate \(album\) artists and live tv channel rows to SDK #3647, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Replace ItemQuery with GetItemsRequest #3654, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate "Play first unwatched" to SDK #3655, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Send QueueEntry to playback backend #3657, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use SDK for FilterOptions #3665, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Move media session implementation to exoplayer module #3670, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Migrate Live TV code to SDK #3691, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Improve intro loading in SdkPlaybackHelper #3755, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove unused compat streaminfo code #3762, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Replace all left\-over Date usages with java\.time #3767, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Split "playback/exoplayer" Gradle module into two #3781, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## Contributors

- [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- [@pascalwei](https://github.com/pascalwei)
- [@haydenzeller](https://github.com/haydenzeller)
- [@DoggoOfSpeed](https://github.com/DoggoOfSpeed)
- [@bathompson](https://github.com/bathompson)
- [@VizuaaLOG](https://github.com/VizuaaLOG)
- [@3l0w](https://github.com/3l0w)
- [@mnsrulz](https://github.com/mnsrulz)
- [@MichaelRUSF](https://github.com/MichaelRUSF)
- [@efreet7](https://github.com/efreet7)
- [@ConnorS1110](https://github.com/ConnorS1110)
- [@cbeyls](https://github.com/cbeyls)
- [@Bond\-009](https://github.com/Bond\-009)

<hr /><em>This discussion was created from the release <a href='https://github.com/jellyfin/jellyfin-androidtv/releases/tag/v0.17.0'>v0.17.0</a>.</em>

**Comment by javierzhang0504** [comment]
Is there any plan to add mpv as native player ?

**Reply by nielsvanvelzen** [reply]
Could you elaborate as to why you'd like to have an MPV backend?

**Reply by javierzhang0504** [reply]
> Could you elaborate as to why you'd like to have an MPV backend?

because of the ssa subtitle.
Now HDR+ssa will forced transcode to SDR

**Reply by nielsvanvelzen** [reply]
Right that's a bit of a problem. We need to bake in SSA because our video engine (media3.exoplayer) does not currently work correctly with those subtitles. Unfortunately the server does not currently support HDR to HDR transcoding, which would fix that part of the issue.

I'm looking into adding good support for SSA myself to our video player because that's the most significant reason people used the experimental (and broken) LibVLC implementation in previous versions.

More information about why the libVLC player was can be read in the pull request description: #3639

**Reply by javierzhang0504** [reply]
> Right that's a bit of a problem. We need to bake in SSA because our video engine (media3.exoplayer) does not currently work correctly with those subtitles. Unfortunately the server does not currently support HDR to HDR transcoding, which would fix that part of the issue.
> 
> 
> 
> I'm looking into adding good support for SSA myself to our video player because that's the most significant reason people used the experimental (and broken) LibVLC implementation in previous versions.
> 
> 
> 
> More information about why the libVLC player was can be read in the pull request description: #3639

Thank you for your answer!

**Comment by Befbo** [comment]
I used to use the "ends at" function while browsing to figure out if I had enough time to watch something. It looks like it was removed here https://github.com/jellyfin/jellyfin-androidtv/pull/3724 . I realize you can go into the entry and see this info, but it was nice to be able to see while clicking through options. Maybe just a preference option for how you wish to display this info in the settings?

**Reply by nielsvanvelzen** [reply]
Over time more and more information was added to that row and we had to cut some information so it wouldn't overflow. I've opted for removing the ends at part because it already shows the duration and there is a clock on the top-right of the app, so it's a bit redundant.
I'm not opposed to show the "ends at" part again but we need to find space to do that first.

**Reply by Befbo** [reply]
That's fair. Thanks for your work!

**Reply by Nickbert7** [reply]
I also liked the information how it was shown in the past.
Could it may be added below the clock in the corner in a smaller font?
So you see the current time and the end time in one catch? (Maybe as an option in the settings as there are already settings for the clock)
Something like this:
![image](https://github.com/user-attachments/assets/26ab3dbe-0c10-472b-85ec-b60e234c4316)

**Reply by nielsvanvelzen** [reply]
One problem with that is that the alignment of elements doesn't allow for the "info row" to go beneath the top-right part of the app

![92000003b5](https://github.com/user-attachments/assets/c4caf27a-0f23-4253-bb20-84e1646624ff)

So it would require some additional changes to bring that bar down a bit etc. Nothing unsolvable but a little bit more involved.

**Comment by Alliyance** [comment]
Any idea if anyone is working on the Subtitles issue? 
[https://github.com/jellyfin/jellyfin/issues/11776](url)

By the way, A big thank you for resetting the Audio direct stream.:)

**Comment by Kuriix** [comment]
Now that LibVLC is removed I can't edit Audio/Subtitles delay anymore. Is there any setting to enable this function?

**Reply by nielsvanvelzen** [reply]
See #2479

**Comment by TrentonBlackburn** [comment]
Hey there! I chat occasionally on the discord server, but was trying to find the best spot to formalize a request:

Is there any way we could have a **toggle option for using logo artwork**? I know it was added in 0.17.0, but I find that the sizing and sometimes silliness of the default logos are really inconsistent. I don't have pictures right now as examples, and I know I could change the logo artwork on an individual basis to better suit my needs... but I'm wondering if we could possibly also just retain the old legacy implementation.

I think just the simple white text looked really great, and matched with the "Jellyfin" and Timestamp in the top corners of the screensaver. I'm really happy to see the age filter so my entire library's artwork can now show up, but I think this logo toggle would crystallize the screensaver as a perfect feature IMO. 

Thanks!

**Reply by nielsvanvelzen** [reply]
I don't see a point in preferences for tiny changes like this one. It makes the code harder to maintain and introduces an additional path for testing.

If you don't want logo's to show up in Jellyfin you should remove them. If they don't look good, you should fix them.

**Reply by TrentonBlackburn** [reply]
I appreciate how much work you do on this repo, but I would argue that both styles should be supported.

Some people have very large catalogues of media, and are likely using the default assets for backdrop and logo. I have only changed maybe 3% of my logo assets manually, and the inconsistency of logo sizing and styling is resulting in a worse experience than the original implementation in my opinion. 

There are also some logo assets that don't have a size/color/style option that would match up with other titles. I think it's a valid example of a setting to either have a consistent yet basic title, or a potentially inconsistent yet "richer" PNG logo option. 

In a vacuum using the custom assets is better, but in practice it is hindered because there is no stylistic consistency in the imported default assets. 

I'm a .NET developer and haven't touched Android in a long time so it would take me a hot minute to get a working environment set up, but if this is something that won't be considered from you, would you be open to a pull request for it potentially?

I understand that essentially having a big conditional statement wrapping 2 completely different implementations isn't the prettiest code on earth, but that's why they are called "preferences", no?
